src/cheevos/Cheevos.cpp

```diff
@@ -0,0 +1,285 @@
+/*
+ *  This file is based on https://github.com/libretro/RetroArch/blob/96c5f5dfb07454c972c50838815330382d6b1911/cheevos/fixup.c
+ *
+ *  Copyright (C) 2014-2020 Team Kodi (https://kodi.tv)
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSE.md for more information.
+ */
+
+#include "Cheevos.h"
+
+#include "libretro/LibretroEnvironment.h"
+#include "libretro/MemoryMap.h"
+#include "libretro/libretro.h"
+#include "rcheevos/rconsoles.h"
+#include "rcheevos/rhash.h"
+#include "rcheevos/rurl.h"
+
+using namespace LIBRETRO;
+
+namespace
+{
+constexpr int HASH_SIZE = 33;
+constexpr int URL_SIZE = 512;
+constexpr int RICH_PRESENCE_EVAL_SIZE = 512;
+constexpr int POST_DATA_SIZE = 1024;
+} // namespace
+
+CCheevos::CCheevos()
+{
+  rc_runtime_init(&m_runtime);
+}
+
+void CCheevos::Initialize()
+{
+  rc_runtime_init(&m_runtime);
+}
+
+void CCheevos::Deinitialize()
+{
+  rc_runtime_destroy(&m_runtime);
+}
+
+void CCheevos::ResetRuntime()
+{
+  rc_runtime_reset(&m_runtime);
+  m_richPresence =
+      rc_parse_richpresence(m_richPresenceBuffer.data(), m_richPresenceScript.c_str(), NULL, 0);
+}
+
+CCheevos& CCheevos::Get(void)
+{
+  static CCheevos _instance;
+  return _instance;
```

Comment:
Singletons are an anti-feature because they inhibit the ability to have multiple instances if needed later. It would be better if this object was a child of another object, like the environment or the add-on instance. However, singletons are the current pattern left over from the old add-on API, so merge this and I'll refactor later.